### PR TITLE
normalise updates

### PR DESCRIPTION
**Pull request by surgan12**
enhancement  #680.

### CODE REVIEW

**Comment by surgan12** [comment]
@fmassa  would love to hear from you on this !

File: torchvision/transforms/functional.py [hunk 1]
```diff
@@ -200,10 +200,18 @@ def normalize(tensor, mean, std):
...
+    if not inplace:
+        tensor_clone = tensor.clone()
+        # This is faster than using broadcasting, don't change without benchmarking
+        for t, m, s in zip(tensor_clone, mean, std):
+            t.sub_(m).div_(s)
+        return tensor_clone
+
+    else:
+        # This is faster than using broadcasting, don't change without benchmarking
+        for t, m, s in zip(tensor, mean, std):
```
Comment (fmassa) [inline torchvision/transforms/functional.py]: I believe this can now be replaced with something simpler and as efficient as before
```python
mean = torch.tensor(mean, dtype=torch.float32)
std = torch.tensor(std, dtype=torch.float32)
return (tensor - mean[:, None, None]) / std[:, None, None]
```
This is out-of-place, but can be made in-place as well if needed


Just to double check, can you do some benchmarkings and report back if it's indeed now giving the same runtime speed?

File: torchvision/transforms/functional.py [hunk 2]
```diff
@@ -200,10 +200,18 @@ def normalize(tensor, mean, std):
     if not _is_tensor_image(tensor):
         raise TypeError('tensor is not a torch image.')
 
-    # This is faster than using broadcasting, don't change without benchmarking
-    for t, m, s in zip(tensor, mean, std):
-        t.sub_(m).div_(s)
-    return tensor
+    if not inplace:
+        tensor_clone = tensor.clone()
```
Comment (fmassa) [inline torchvision/transforms/functional.py]: Can't you just do
```python
if not inplace:
    tensor = tensor.clone()
# everything as before
```
?

File: torchvision/transforms/functional.py (hunk 1 above)
Comment (surgan12) [reply torchvision/transforms/functional.py]: @fmassa  i ran some tests and found that this approach was smewhat faster .

File: torchvision/transforms/functional.py [hunk 3]
```diff
@@ -200,10 +200,17 @@ def normalize(tensor, mean, std):
     if not _is_tensor_image(tensor):
         raise TypeError('tensor is not a torch image.')
 
-    # This is faster than using broadcasting, don't change without benchmarking
-    for t, m, s in zip(tensor, mean, std):
-        t.sub_(m).div_(s)
-    return tensor
+    if not inplace:
+        tensor = tensor.clone()
```
Comment (fmassa) [inline torchvision/transforms/functional.py]: Can we just have a single clone here, to avoid duplicated code?
like
```python
if not inplace:
    tensor = tensor.clone()

mean = ...
```
